packages/lib/gpcircuits/test/bounds.spec.ts

```diff
@@ -0,0 +1,268 @@
+import { WitnessTester } from "circomkit";
+import "mocha";
+import {
+  BoundsCheckModuleInputNamesType,
+  BoundsCheckModuleOutputNamesType
+} from "../src";
+import { circomkit } from "./common";
+
+function inInterval(input: bigint, bounds: bigint[]): boolean {
+  return bounds[0] <= input && input <= bounds[1];
+}
+
+// 'Critical intervals' are those intervals of field elements which are larger
+// than `nBits` bits yet still satisfy the constraints of the `InInterval`
+// circuit for given bounds [a,b].  These are the intervals [1 << nBits, 1 <<
+// nBits + a - 1] and [p - (1 << nBits) + b + 1, p-1], where p is the field
+// modulus. This procedure returns these as an array of closed intervals, where
+// empty intervals are omitted.
+function criticalIntervals(
```

Comment:
This is cool!  I look forward to the mathematical argument for why these critical intervals are the right ones to test.